ratik/src/intTest/java/org/opencadc/ratik/InventoryValidatorTest.java

```diff
@@ -0,0 +1,855 @@
+        Assert.assertNotNull("DeletedArtifactEvent not found", deletedArtifactEvent);
+
+        // case 2: local contentModifiedDate after remote contentModifiedDate
+        contentModifiedDate = Calendar.getInstance();
+
+        remoteArtifact = new Artifact(remoteID, URI.create("cadc:INTTEST/one.ext"), contentCheckSum,
+                                      contentModifiedDate.getTime(), 1024L);
+        this.remoteEnvironment.artifactDAO.put(remoteArtifact);
+
+        contentModifiedDate.roll(Calendar.YEAR, true);
+        localArtifact = new Artifact(localID, URI.create("cadc:INTTEST/one.ext"), contentCheckSum,
+                                     contentModifiedDate.getTime(), 1024L);
+        this.localEnvironment.artifactDAO.put(localArtifact);
+
+        try {
+            System.setProperty("user.home", TMP_DIR);
+            InventoryValidator testSubject = new InventoryValidator(this.localEnvironment.daoConfig, TestUtil.LUSKAN_URI,
+                                                                    new IncludeArtifacts(),null,
+                                                                    trackSiteLocations);
+            testSubject.run();
+        } finally {
+            System.setProperty("user.home", USER_HOME);
+        }
+
+        localArtifact = this.localEnvironment.artifactDAO.get(localArtifact.getID());
+        Assert.assertNotNull("local artifact not found", localArtifact);
+
+        deletedArtifactEvent = this.localEnvironment.deletedArtifactEventDAO.get(localID);
+        Assert.assertNull("DeletedArtifactEvent found", deletedArtifactEvent);
+    }
+
+    /**
+     * discrepancy: artifact in both && valid metaChecksum mismatch
+     *
+     * explanation1: pending/missed artifact update in L
+     * evidence: ??
+     * action: put Artifact
+     *
+     * before: Artifact in L & R, update Artifact in R
+     * after: R Artifact in L
+     *
+     * explanation2: pending/missed artifact update in R
+     * evidence: ??
+     * action: do nothing
+     *
+     * before: Artifact in L & R, update Artifact in L
+     * after: Artifact in L
+     */
+    @Test
+    public void artifactChecksumMismatch_Explanation1_LocalIsStorage() throws Exception {
+        artifactChecksumMismatch_Explanation1(false);
+    }
+
+    @Test
+    public void artifactChecksumMismatch_Explanation1_LocalIsGlobal() throws Exception {
+        artifactChecksumMismatch_Explanation1(true);
+    }
+
+    public void artifactChecksumMismatch_Explanation1(boolean trackSiteLocations) throws Exception {
+        UUID artifactID = UUID.randomUUID();
+        URI artifactURI = URI.create("cadc:INTTEST/one.ext");
+
+        Artifact artifact = new Artifact(artifactID, artifactURI, TestUtil.getRandomMD5(),
+                                              new Date(), 1024L);
+        this.localEnvironment.artifactDAO.put(artifact);
+        this.remoteEnvironment.artifactDAO.put(artifact);
+
```

Comment:
Since "pick a winner" is likely to be based on relative Entity.lastModified, I would put a short sleep in here so the updated artifact has a reliably different timestamp.... for someone testing locally with two local container database servers the clocks are the same so 50ms is plenty of sleep. (if the db servers are on different hosts, clocks can be off by ~minutes and that's a real pita to test)

Reply:
Hadn't thought of that, added a sleep between the artifact puts.